examples/mxnet_imagenet_resnet50.py

```diff
@@ -0,0 +1,287 @@
+    rec_train = os.path.expanduser(rec_train)
+    rec_train_idx = os.path.expanduser(rec_train_idx)
+    rec_val = os.path.expanduser(rec_val)
+    rec_val_idx = os.path.expanduser(rec_val_idx)
+    jitter_param = 0.4
+    lighting_param = 0.1
+    mean_rgb = [123.68, 116.779, 103.939]
+    std_rgb = [58.393, 57.12, 57.375]
+
+    def batch_fn(batch, ctx):
+        data = gluon.utils.split_and_load(batch.data[0], ctx_list=ctx, batch_axis=0)
+        label = gluon.utils.split_and_load(batch.label[0], ctx_list=ctx, batch_axis=0)
+        return data, label
+
+    train_data = mx.io.ImageRecordIter(
+        path_imgrec         = rec_train,
+        path_imgidx         = rec_train_idx,
+        preprocess_threads  = data_nthreads,
+        shuffle             = True,
+        batch_size          = batch_size,
+        label_width         = 1,
+        data_shape          = (3, 224, 224),
+        mean_r              = mean_rgb[0],
+        mean_g              = mean_rgb[1],
+        mean_b              = mean_rgb[2],
+        rand_mirror         = True,
+        rand_crop           = False,
+        random_resized_crop = True,
+        max_aspect_ratio    = 4. / 3.,
+        min_aspect_ratio    = 3. / 4.,
+        max_random_area     = 1,
+        min_random_area     = 0.08,
+        verbose             = False,
+        brightness          = jitter_param,
+        saturation          = jitter_param,
+        contrast            = jitter_param,
+        pca_noise           = lighting_param,
+        num_parts           = hvd.size(),
+        part_index          = hvd.rank()
+    )
+    # kept each node to use full val data to make it easy to monitor results
+    val_data = mx.io.ImageRecordIter(
+        path_imgrec         = rec_val,
+        path_imgidx         = rec_val_idx,
+        preprocess_threads  = data_nthreads,
+        shuffle             = False,
+        batch_size          = batch_size,
+        resize              = 256,
+        label_width         = 1,
+        rand_crop           = False,
+        rand_mirror         = False,
+        data_shape          = (3, 224, 224),
+        mean_r              = mean_rgb[0],
+        mean_g              = mean_rgb[1],
+        mean_b              = mean_rgb[2]
+    )
+
+    return train_data, val_data, batch_fn
+
+# Create data iterator for synthetic data
+class SyntheticDataIter(DataIter):
+    def __init__(self, num_classes, data_shape, max_iter, dtype):
+        self.batch_size = data_shape[0]
+        self.cur_iter = 0
+        self.max_iter = max_iter
+        self.dtype = dtype
+        label = np.random.randint(0, num_classes, [self.batch_size,])
+        data = np.random.uniform(-1, 1, data_shape)
+        self.data = mx.nd.array(data, dtype=self.dtype, ctx=mx.Context('cpu_pinned', hvd.local_rank()))
+        self.label = mx.nd.array(label, dtype=self.dtype, ctx=mx.Context('cpu_pinned', hvd.local_rank()))
+    def __iter__(self):
+        return self
+    @property
```

Comment:
Style: can you format according to PEP-8?  There should be a space between member declarations, two spaces around classes, etc.

Reply:
updated